app/routes.py

```diff
@@ -1,2 +1,323 @@
+        elif sort_query == "postal_code":
+            customers = Customer.query.order_by(asc(Customer.customer_zip))
+        elif sort_query == "registered_at":
+            customers = Customer.query.order_by(asc(Customer.register_at))
+        elif sort_query == "id":
+            customers = Customer.query.order_by(asc(Customer.customer_id))
+        else:
+            return make_400()
+    else:
+        customers = Customer.query.all()
+    customer_response = []
+    for customer in customers:
+        customer_response.append(customer.to_json())    
+    return jsonify(customer_response), 200
+
+@customer_bp.route("", methods=["DELETE"])
+def delete_all_customers():
+    customers = Customer.query.all()
+    for customer in customers:
+        db.session.delete(customer)
+    db.session.commit()
+    return make_response("All of the customers in the database have been deleted.", 200)
+
+@customer_bp.route("/<customer_id>", methods=["GET"])
+def get_single_customer(customer_id):
+    customer = Customer.query.get(customer_id)
+    if customer is None:
+        return make_404()
+    else:
+        return make_response(customer.to_json(), 200)
+
+@customer_bp.route("/<customer_id>", methods=["PUT"])
+def update_single_customer(customer_id):
+    customer = Customer.query.get(customer_id)
+    if customer is None:
+        return make_404()
+    else:
+        request_body = request.get_json()
+        try:
+            customer.customer_name = request_body["name"]
+            customer.customer_zip = request_body["postal_code"]
+            customer.customer_phone = request_body["phone"]
+        except KeyError:
+            return make_400()
+        db.session.commit()
+        return make_response(customer.to_json(), 200)
+
+@customer_bp.route("", methods=["POST"])
+def post_new_customer():
+    request_body = request.get_json()
+    try:
+        new_customer = Customer.new_customer_from_json(request_body)
+    except KeyError:
+        return make_400()
+    db.session.add(new_customer)
+    db.session.commit()
+    return make_response({"id": new_customer.customer_id}, 201)
+
+@customer_bp.route("/<customer_id>", methods=["DELETE"])
+def delete_single_customer(customer_id):
+    customer = Customer.query.get(customer_id)
+    if customer:
+        db.session.delete(customer)
+        db.session.commit()
+    else:
+        return make_404()
+    return make_response({"id": customer.customer_id}, 200)
+
+@customer_bp.route("/<customer_id>/rentals", methods=["GET"])
+def get_customers_rentals(customer_id):
+    customer_info = get_rental_by_customer(customer_id)
+    rental_list = []
+    for tuple_set in customer_info:
+        video = tuple_set[1]
+        rental = tuple_set[2]
+        rental_list.append({
+            "release_date": video.release_date, 
+            "title": video.video_title, 
+            "due_date": rental.due_date
+            })
+    return jsonify(rental_list), 200
```

Comment:
You can refactor this using relationships:

```suggestion
    customer = Customer.query.get_or_404(customer_id)
    rental_list = []
    for rental in customer.rentals:
        rental_list.append({
            "release_date": rental.video.release_date, 
            "title": rental.video.video_title, 
            "due_date": rental.due_date
            })
    return jsonify(rental_list), 200
```